### DISCUSSION TITLE
Does anybody know custom nodes for pixel-perfect image mirroring and 90° rotation?

### DISCUSSION THREAD

**Discussion by aleksusklim** [Comfy-Org/ComfyUI#1642] [Q&A]
We have latent flip and latent rotate (by 90°), now I need to rotate the rendered image.

I saw https://github.com/Nourepide/ComfyUI-Allor#rotate, but looks like it uses filter to process images, while I need pixel-perfect 90° rotation and mirroring.
Does anybody know custom node which already does this?

**Comment by polarsky89** [comment]
Any luck in this matter?

**Reply by aleksusklim** [reply]
Turns out latent flip or rotation DESTROYS the picture!

This is the original:
![0p__00001_](https://github.com/comfyanonymous/ComfyUI/assets/13969463/0e3e636a-b312-4ce8-a755-10c3b448c903)

And here is how latent rotate + image rotate back (what I wanted to do automatically but had to do it manually in image editor) looks like:
![1p__00001_](https://github.com/comfyanonymous/ComfyUI/assets/13969463/b490c839-6ae6-464e-a0f2-9c64c143aef0)
![2p__00001_](https://github.com/comfyanonymous/ComfyUI/assets/13969463/3da28565-53cb-48b3-8d27-4744e60c5162)
![3p__00001_](https://github.com/comfyanonymous/ComfyUI/assets/13969463/d30446c8-3f11-4851-8151-3f85fc4cde4f)

Latent-fipped versions:
![4m__00001_](https://github.com/comfyanonymous/ComfyUI/assets/13969463/642d8288-b257-40fd-8509-2fed4b1c22d9)
![5m__00001_](https://github.com/comfyanonymous/ComfyUI/assets/13969463/7a434a16-ebbb-4229-91b8-775f0050a107)
![6m__00001_](https://github.com/comfyanonymous/ComfyUI/assets/13969463/1d3bbcb1-9501-4000-bfe7-ac5b8f2462f5)
![7m__00001_](https://github.com/comfyanonymous/ComfyUI/assets/13969463/2dd64d7a-1f6a-4f8f-9a02-a826caba956b)
